src/components/ha-dialog-date-picker.ts

```diff
@@ -56,7 +56,7 @@ export class HaDialogDatePicker extends LitElement {
   }
 
   private _setToday() {
-    this._value = new Date().toISOString().split("T")[0];
+    this._value = new Date().toLocaleDateString("en-CA");
```

Comment:
Feels a bit weird to use `en-CA` maybe just use `${today.getFullYear()}-${today.getMonth()+1}-${today.getDate()}`?

Reply:
I did consider that, but we end up having to pad the month and date to be two digits as well. It does feel weird to use a specific locale for it.

Reply:
Another option is to keep using the ISOString, but add `getTimezoneOffset() * 60 * 1000` before...

Reply:
> Another option is to keep using the ISOString, but add `getTimezoneOffset() * 60 * 1000` before...

All of these options feel like a "hack"/not ideal. Even with the ISOString, we are still hacking off the T##:##:##PM from the end of it. 
Although having "en-CA" could make it confusing for people in the future to interpret this code. Maybe a comment is appropriate in the code that en-CA is used to get the locale format of YYYY-MM-DD.

Personally I think we should leave it at "en-CA", but I can change it to ISOString with timezone offset if you'd like.

Reply:
Ok, let's leave it at `en-CA`, it seems browser support should be just about there...

Reply:
Added https://github.com/home-assistant/frontend/pull/14043/commits/5f863d10a05b1eece107d0f9bf21e58020f0550b